### PR TITLE
Add 'delete nodepool' command

### PR DESCRIPTION
**Pull request by marians**
Towards https://github.com/giantswarm/giantswarm/issues/6447

This PR adds the `gsctl delete nodepool` command.

### Usage preview

```
Delete new node pool from a cluster.

This command allows to delete a node pool.

Deleting a node pool means that all worker nodes in the pool will be drained,
cordoned and then terminated.

In case you are running workloads on the node pool you want to delete,
make sure that there is at least one other node pool with capacity to
schedule the workloads. Also check whether label selectors, taints and
tolerations will allow scheduling on other pool's worker nodes. The best
way to observe this is by manually cordoning and draining the pool's
worker nodes and checking workload's node assignments, before issuing
the 'delete nodepool' command.

Note: Data stored outside of persistent volumes will be lost and there is
no way to undo this.

Examples:

  To delete node pool 'np1id' from cluster 'f01r4', use this command:

    gsctl delete nodepool f01r4/np1id

  To prevent the confirmation questions, apply --force:

    gsctl delete nodepool f01r4/np1id --force

Usage:
  gsctl delete nodepool <cluster-id>/<nodepool-id> [flags]

Aliases:
  nodepool, np

Flags:
      --force   If set, no interactive confirmation will be required (risky!).
  -h, --help    help for nodepool
```

## Execution preview

```
$ gsctl delete nodepool m0ckd/abc
Do you really want to delete node pool 'abc' from cluster 'm0ckd'? [y/n]: y
Node pool 'abc' in cluster 'm0ckd' will be deleted as soon as all workloads are terminated.
```

### CODE REVIEW

File: client/clienterror/clienterror.go [hunk 1]
```diff
@@ -333,6 +333,31 @@ func New(err error) *APIError {
...
+			ErrorMessage:   "Unauthorized",
+			ErrorDetails:   "You don't have permission to delete this node pool.",
+		}
+	}
+	if deleteNodePoolsNotFoundErr, ok := err.(*nodepools.DeleteNodePoolNotFound); ok {
+		return &APIError{
+			HTTPStatusCode: http.StatusNotFound,
+			OriginalError:  deleteNodePoolsNotFoundErr,
+			ErrorMessage:   "Not found",
+			ErrorDetails:   "The cluster or node pool was not found or you don't have access to it.",
```
Comment (xh3b4sd) [inline client/clienterror/clienterror.go]: The other errors exclusively talk about node pools. This here mentions clusters as well. Is this how it should be?

File: commands/create/nodepool/command.go [hunk 2]
```diff
@@ -9,12 +9,13 @@ import (
 	"github.com/fatih/color"
 	"github.com/giantswarm/gscliauth/config"
 	"github.com/giantswarm/gsclientgen/models"
+	"github.com/giantswarm/microerror"
+	"github.com/spf13/cobra"
```
Comment (xh3b4sd) [inline commands/create/nodepool/command.go]: This is very cool, but one of the instances in which we could have take a step back, and create a separate PR.

File: commands/delete/nodepool/command.go [hunk 3]
```diff
@@ -0,0 +1,216 @@
...
+	clientWrapper, err := client.NewWithConfig(args.APIEndpoint, args.UserProvidedToken)
+	if err != nil {
+		return false, microerror.Mask(err)
+	}
+
+	auxParams := clientWrapper.DefaultAuxiliaryParams()
+	auxParams.ActivityName = activityName
+
+	_, err = clientWrapper.DeleteNodePool(args.ClusterID, args.NodePoolID, auxParams)
+	if err != nil {
```
Comment (xh3b4sd) [inline commands/delete/nodepool/command.go]:
```
if IsError(err) {
  //
} else if IsError(err) {
  // 
} else if err != nil {
  // 
}
```

File: commands/delete/nodepool/command.go [hunk 4]
```diff
@@ -0,0 +1,216 @@
...
+		if subtext != "" {
+			fmt.Println(subtext)
+		}
+		os.Exit(1)
+	}
+
+	if deleted {
+		fmt.Println(color.GreenString("Node pool '%s' in cluster '%s' will be deleted as soon as all workloads are terminated.", args.NodePoolID, args.ClusterID))
+	} else if args.Verbose {
+		fmt.Println(color.WhiteString("Aborted."))
```
Comment (xh3b4sd) [inline commands/delete/nodepool/command.go]: So we get the full stop here now. Why is that FMI? I have been told single word labels, titles, responses do not need that.

File: commands/delete/nodepool/command_test.go [hunk 5]
```diff
@@ -0,0 +1,283 @@
...
+				APIEndpoint: "https://foo",
+				AuthToken:   "some-token",
+				ClusterID:   "clusterid",
+				NodePoolID:  "nodepoolid",
+				Force:       true,
+			},
+		},
+	}
+
+	yamlText := `last_version_check: 0001-01-01T00:00:00Z
```
Comment (xh3b4sd) [inline commands/delete/nodepool/command_test.go]: Can we define that outside the function to make the test easier to read?

File: commands/delete/nodepool/command_test.go [hunk 6]
```diff
@@ -0,0 +1,283 @@
...
+	"testing"
+
+	"github.com/giantswarm/gsctl/commands/errors"
+	"github.com/giantswarm/gsctl/testutils"
+	"github.com/google/go-cmp/cmp"
+	"github.com/spf13/afero"
+)
+
+// TestCollectArgs tests whether collectArguments produces the expected results.
+func TestCollectArgs(t *testing.T) {
```
Comment (xh3b4sd) [inline commands/delete/nodepool/command_test.go]:
```suggestion
func Test_collectArguments(t *testing.T) {
```

File: commands/delete/nodepool/command_test.go [hunk 7]
```diff
@@ -0,0 +1,283 @@
...
+	"strconv"
+	"testing"
+
+	"github.com/giantswarm/gsctl/commands/errors"
+	"github.com/giantswarm/gsctl/testutils"
+	"github.com/google/go-cmp/cmp"
+	"github.com/spf13/afero"
+)
+
+// TestCollectArgs tests whether collectArguments produces the expected results.
```
Comment (xh3b4sd) [inline commands/delete/nodepool/command_test.go]:
```suggestion
// Test_collectArguments tests whether collectArguments produces the expected results.
```

File: commands/delete/nodepool/command_test.go [hunk 8]
```diff
@@ -0,0 +1,283 @@
...
+				t.Errorf("Case %d - Unexpected error '%s'", i, err)
+			}
+			if diff := cmp.Diff(tc.resultingArgs, args); diff != "" {
+				t.Errorf("Case %d - Resulting args unequal. (-expected +got):\n%s", i, diff)
+			}
+		})
+	}
+}
+
+// TestVerifyPreconditions tests cases where validating preconditions fails.
```
Comment (xh3b4sd) [inline commands/delete/nodepool/command_test.go]:
```suggestion
// Test_verifyPreconditions tests cases where validating preconditions fails.
```

File: commands/delete/nodepool/command_test.go [hunk 9]
```diff
@@ -0,0 +1,283 @@
...
+func TestVerifyPreconditions(t *testing.T) {
```
Comment (xh3b4sd) [inline commands/delete/nodepool/command_test.go]:
```suggestion
func Test_verifyPreconditions(t *testing.T) {
```

File: client/clienterror/clienterror.go (hunk 1 above)
Comment (marians) [reply client/clienterror/clienterror.go]: The thinking is that a 404 error could mean either that the node pool does not exist OR that the owning cluster does not exist. We don't know.

File: commands/delete/nodepool/command.go (hunk 4 above)
Comment (marians) [reply commands/delete/nodepool/command.go]: Probably you crept into my head and changed my thinking. Can't tell. Humans.

**Comment by marians** [comment]
Done? PTAL @xh3b4sd

**Pull request merged**